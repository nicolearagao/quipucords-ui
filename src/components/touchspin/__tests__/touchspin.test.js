import React from 'react';
import { TouchSpin } from '../touchspin';

describe('TouchSpin Component', () => {
  it('should render a basic component', () => {
    const props = {
      name: 'lorem'
    };

    const component = renderComponent(<TouchSpin {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should handle onchange events', () => {
    const props = {
      name: 'lorem',
      value: 1,
      labelMaxDescription: 'testing max',
      labelMinDescription: 'testing min'
    };

    const component = renderComponent(<TouchSpin {...props} />);

    const inputMax = component.find(`button[aria-label="${props.labelMaxDescription}"]`);
    component.fireEvent.click(inputMax);
    expect(component.find('input').value).toBe('2');

    const inputMin = component.find(`button[aria-label="${props.labelMinDescription}"]`);
    component.fireEvent.click(inputMin);
    expect(component.find('input').value).toBe('1');
  });
});
