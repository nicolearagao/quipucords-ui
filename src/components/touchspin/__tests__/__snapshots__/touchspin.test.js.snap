// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchSpin Component should render a basic component: basic 1`] = `
<touchspin
  props="{
  "name": "lorem",
  "className": "",
  "labelMax": {
    "key": null,
    "ref": null,
    "props": {
      "color": "currentColor",
      "size": "sm",
      "noVerticalAlign": false
    },
    "_owner": null,
    "_store": {}
  },
  "labelMaxDescription": null,
  "labelMin": {
    "key": null,
    "ref": null,
    "props": {
      "color": "currentColor",
      "size": "sm",
      "noVerticalAlign": false
    },
    "_owner": null,
    "_store": {}
  },
  "labelMinDescription": null,
  "maxValue": 100,
  "minValue": 0,
  "value": 50,
  "children": []
}"
>
  <div
    class="pf-c-number-input "
  >
    <div
      class="pf-c-input-group"
    >
      <button
        aria-disabled="false"
        aria-label="t(form-dialog.label_touchspin, {"context":"min"})"
        class="pf-c-button pf-m-control"
        data-ouia-component-id="OUIA-Generated-Button-control-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
      <input
        aria-invalid="false"
        aria-label="t(form-dialog.label_touchspin, {"context":"input"})"
        class="pf-c-form-control"
        data-ouia-component-id="OUIA-Generated-TextInputBase-1"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        name="lorem"
        type="number"
        value="50"
      />
      <button
        aria-disabled="false"
        aria-label="t(form-dialog.label_touchspin, {"context":"max"})"
        class="pf-c-button pf-m-control"
        data-ouia-component-id="OUIA-Generated-Button-control-2"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </div>
  </div>
</touchspin>
`;
